src/index.ts

```diff
@@ -103,12 +103,16 @@ const setupServer = async () => {
     }
   };
   const server = new ApolloServer({ typeDefs, resolvers });
-  server.listen(process.env.PORT).then(({ url }) => {
-    console.log(`🚀  Server ready at ${url}`);
-  });
+  try {
+    await server.listen(process.env.PORT).then(({ url }) => {
+      console.log(`🚀  Server ready at ${url}`);
+    });
+  } catch {
+    (error) => console.error(error);
+  }
```

Comment:
```suggestion
  const { url } = await server.listen(process.env.PORT);
  console.log(`🚀  Server ready at ${url}`);
```

2 pontos aqui:

1. Não precisa do then quando vc usa await, porque é só coletar o resultado numa variável e usar abaixo.
2. O try-catch vai fazer com que não estoure erro, mas aqui se o `listen` não der certo, acho que seria o caso de deixar estourar (geralmente não deixamos mesmo, seria um caso de exceção) porque se ele não conseguiu subir o server GraphQL, nem adianta continuar, certo?